plugins/status/plugin_test.go

```diff
@@ -230,6 +232,53 @@ func TestPluginReconfigure(t *testing.T) {
 	}
 }
 
+func TestMetrics(t *testing.T) {
+	metrics.SetProvider("prometheus", nil)
+	fixture := newTestFixture(t)
+	fixture.server.ch = make(chan UpdateRequestV1)
+	fixture.plugin.config.IncludeMetrics = true
+	defer fixture.server.stop()
+
+	ctx := context.Background()
+
+	fixture.plugin.Start(ctx)
+	defer fixture.plugin.Stop(ctx)
+
+	status := testStatus()
+
+	fixture.plugin.BulkUpdateBundleStatus(map[string]*bundle.Status{"bundle": status})
+	result := <-fixture.server.ch
+	// check some well-known metrics that must be present in the result
+
+	metricsBytes, err := json.Marshal(result.Metrics["prometheus"])
+	if err != nil {
+		t.Fatal(err)
+	}
+	var prometheusMetrics []struct {
+		Name   string
+		Metric []interface{}
+	}
+
+	if err := json.Unmarshal(metricsBytes, &prometheusMetrics); err != nil {
+		t.Fatal(err)
+	}
+
+	index := -1
+	for i, name := range []string{"go_info", "go_threads", "go_goroutines"} {
```

Comment:
nit: IMO these tests for prometheus specific bits should be in the prometheus package testing its `Gather` API. The tests for the status package should use the dummy or a mock metrics provider.

Reply:
Will change to mock provider